lib/s3routes/routesUtils.js

```diff
@@ -512,7 +516,7 @@ const routesUtils = {
             okContentHeadersResponse(overrideParams, resHeaders, response,
                 range, log);
         }
-        if (dataLocations === null) {
+        if (dataLocations === null || _computeContentLengthFromLocation(dataLocations) === 0) {
```

Comment:
Doesn't `_computeContentLengthFromLocation` return a bool? Wouldn't we want to compare to true or false?

Reply:
nope _contentLengthMatchesLocations return a bool :)

Reply:
whoop yea you right, got them mixed up